app/Services/StandService.php

```diff
@@ -542,46 +544,110 @@ private function standOccupied(NetworkAircraft $aircraft, Stand $stand): bool
         return $distanceFromStand < self::MAX_OCCUPANCY_DISTANCE_METERS;
     }
 
-    /**
-     * Use the stand assignment rules to allocate a stand for a given aircraft
-     */
-    public function allocateStandForAircraft(NetworkAircraft $aircraft): ?StandAssignment
+    public function allocateStandsForArrivals(): void
     {
-        if (!$this->shouldAllocateStand($aircraft)) {
-            return null;
-        }
+        // Get all the eligible aircraft and assign them a stand
+        $onlineAcarsCallsigns = $this->acarsProvider->GetOnlineCallsigns();
+        $sendingAcarsMessages = config('stands.allocation_acars_message', false);
+        $allAirfields = Airfield::all()->mapWithKeys(function (Airfield $airfield) {
+            return [$airfield->code => $airfield];
+        });
 
-        foreach ($this->allocators as $allocator) {
-            if ($allocation = $allocator->allocate($aircraft)) {
-                event(new StandAssignedEvent($allocation));
-                return $allocation;
-            }
-        }
+        $this->getAircraftEligibleForArrivalStandAllocation()
+            ->filter(function (NetworkAircraft $aircraft) use ($allAirfields) {
+                return $this->getTimeFromAirfieldInMinutes($aircraft, $allAirfields[$aircraft->planned_destairport])
+                    < self::ASSIGN_STAND_MINUTES_BEFORE;
+            })
+            ->map(function (NetworkAircraft $aircraft) {
+                foreach ($this->allocators as $allocator) {
+                    if ($allocation = $allocator->allocate($aircraft)) {
+                        return $allocation;
+                    }
+                }
+                return null;
+            })
+            ->filter()
+            ->each(function (StandAssignment $standAssignment) use ($onlineAcarsCallsigns, $sendingAcarsMessages) {
+                event(new StandAssignedEvent($standAssignment));
+                if ($sendingAcarsMessages && $onlineAcarsCallsigns->contains($standAssignment->callsign)) {
```

Comment:
Is any case conversion necessary for the callsigns active on hoppie?

Reply:
I.e. all to the same case to be ultra sure?

Reply:
We probably shouldn't assume, just incase - so probably good to only consider exact matches?